dspace-server-webapp/src/main/java/org/dspace/app/rest/ItemUploadController.java

```diff
@@ -0,0 +1,155 @@
+import org.dspace.app.rest.utils.ContextUtil;
+import org.dspace.app.rest.utils.Utils;
+import org.dspace.authorize.AuthorizeException;
+import org.dspace.content.Bitstream;
+import org.dspace.content.BitstreamFormat;
+import org.dspace.content.Item;
+import org.dspace.content.service.BitstreamFormatService;
+import org.dspace.content.service.BitstreamService;
+import org.dspace.content.service.ItemService;
+import org.dspace.core.Context;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.rest.webmvc.ResourceNotFoundException;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.multipart.MultipartFile;
+
+@RestController
+@RequestMapping("/api/core/items/{uuid}")
+public class ItemUploadController {
+
+    private static final Logger log = LogManager.getLogger();
+
+    @Autowired
+    protected Utils utils;
+
+    @Autowired
+    private ItemService itemService;
+
+    @Autowired
+    private BitstreamService bitstreamService;
+
+    @Autowired
+    private BitstreamConverter bitstreamConverter;
+
+    @Autowired
+    private MetadataConverter metadataConverter;
+
+    @Autowired
+    private BitstreamFormatService bitstreamFormatService;
+
+    @RequestMapping(method = RequestMethod.POST, value = "/bitstreams", headers = "content-type=multipart/form-data")
+    @PreAuthorize("hasPermission(#uuid, 'ITEM', 'WRITE') && hasPermission(#uuid, 'ITEM', 'ADD')")
+    public BitstreamResource uploadBitstream(HttpServletRequest request, @PathVariable UUID uuid,
+                                             @RequestParam("file") MultipartFile uploadfile,
+                                             @RequestParam(value = "properties", required = false) String properties) {
+
+        Context context = ContextUtil.obtainContext(request);
+        Item item = null;
+        Bitstream bitstream = null;
+        try {
+            item = itemService.find(context, uuid);
+        } catch (SQLException e) {
+            log.error("Something went wrong trying to find the Item with uuid: " + uuid, e);
+        }
+        if (item == null) {
+            throw new ResourceNotFoundException("The given uuid did not resolve to an Item on the server: " + uuid);
+        }
+        InputStream fileInputStream = null;
+        try {
+            fileInputStream = uploadfile.getInputStream();
+        } catch (IOException e) {
+            log.error("Something went wrong when trying to read the inputstream from the given file in the request");
+            throw new UnprocessableEntityException("The InputStream from the file couldn't be read");
```

Comment:
The `IOException` here seems to be "swallowed" in that it's not logged or returned to the user.  We need to ensure it is minimally logged in the `log.error` statement here.

Reply:
This has been updated